### Mobile applications analysis

*this project is about mobile apps analysis*

THE GOAL : 

1. Determine apps that are likely to attract more users thus help   developers understand users needs. 
2. Remove innacurate data. 
3. Remove duplicate app entries. 
4. Remove non-English applications

In [12]:
def explore_data(dataset, start, end):
    dataset_slice = dataset[start:end]    
    for row in dataset_slice:
        print(row)
        print('\n') # adds a new (empty) line after each row

    
    print('Number of rows:', len(dataset))
    print('Number of columns:', len(dataset[0]))

In [13]:
from csv import reader
open_file1=open('AppleStore.csv',encoding="utf8")
open_file2=open('googleplaystore.csv',encoding="utf8")
apple=list(reader(open_file1))
google=list(reader(open_file2))

In [15]:
explore_data(apple,1,5)

['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9+', 'Games', '40', '5', '1', '1']


Number of rows: 7198
Number of columns: 16


In [16]:
explore_data(google,1,5)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 10842
Number of columns: 13


In [18]:
freq_tab={}
for app in google:
    cat=app[1]
    if cat not in freq_tab:
        freq_tab[cat]=1
    else:
        freq_tab[cat]+=1
freq_tab


{'Category': 1,
 'ART_AND_DESIGN': 65,
 'AUTO_AND_VEHICLES': 85,
 'BEAUTY': 53,
 'BOOKS_AND_REFERENCE': 231,
 'BUSINESS': 460,
 'COMICS': 60,
 'COMMUNICATION': 387,
 'DATING': 234,
 'EDUCATION': 156,
 'ENTERTAINMENT': 149,
 'EVENTS': 64,
 'FINANCE': 366,
 'FOOD_AND_DRINK': 127,
 'HEALTH_AND_FITNESS': 341,
 'HOUSE_AND_HOME': 88,
 'LIBRARIES_AND_DEMO': 85,
 'LIFESTYLE': 382,
 'GAME': 1144,
 'FAMILY': 1972,
 'MEDICAL': 463,
 'SOCIAL': 295,
 'SHOPPING': 260,
 'PHOTOGRAPHY': 335,
 'SPORTS': 384,
 'TRAVEL_AND_LOCAL': 258,
 'TOOLS': 843,
 'PERSONALIZATION': 392,
 'PRODUCTIVITY': 424,
 'PARENTING': 60,
 'WEATHER': 82,
 'VIDEO_PLAYERS': 175,
 'NEWS_AND_MAGAZINES': 283,
 'MAPS_AND_NAVIGATION': 137,
 '1.9': 1}

In [22]:
del freq_tab['Category']
freq_tab

{'ART_AND_DESIGN': 65,
 'AUTO_AND_VEHICLES': 85,
 'BEAUTY': 53,
 'BOOKS_AND_REFERENCE': 231,
 'BUSINESS': 460,
 'COMICS': 60,
 'COMMUNICATION': 387,
 'DATING': 234,
 'EDUCATION': 156,
 'ENTERTAINMENT': 149,
 'EVENTS': 64,
 'FINANCE': 366,
 'FOOD_AND_DRINK': 127,
 'HEALTH_AND_FITNESS': 341,
 'HOUSE_AND_HOME': 88,
 'LIBRARIES_AND_DEMO': 85,
 'LIFESTYLE': 382,
 'GAME': 1144,
 'FAMILY': 1972,
 'MEDICAL': 463,
 'SOCIAL': 295,
 'SHOPPING': 260,
 'PHOTOGRAPHY': 335,
 'SPORTS': 384,
 'TRAVEL_AND_LOCAL': 258,
 'TOOLS': 843,
 'PERSONALIZATION': 392,
 'PRODUCTIVITY': 424,
 'PARENTING': 60,
 'WEATHER': 82,
 'VIDEO_PLAYERS': 175,
 'NEWS_AND_MAGAZINES': 283,
 'MAPS_AND_NAVIGATION': 137}

In [25]:
name_list=list(freq_tab.keys())
name_list

['ART_AND_DESIGN',
 'AUTO_AND_VEHICLES',
 'BEAUTY',
 'BOOKS_AND_REFERENCE',
 'BUSINESS',
 'COMICS',
 'COMMUNICATION',
 'DATING',
 'EDUCATION',
 'ENTERTAINMENT',
 'EVENTS',
 'FINANCE',
 'FOOD_AND_DRINK',
 'HEALTH_AND_FITNESS',
 'HOUSE_AND_HOME',
 'LIBRARIES_AND_DEMO',
 'LIFESTYLE',
 'GAME',
 'FAMILY',
 'MEDICAL',
 'SOCIAL',
 'SHOPPING',
 'PHOTOGRAPHY',
 'SPORTS',
 'TRAVEL_AND_LOCAL',
 'TOOLS',
 'PERSONALIZATION',
 'PRODUCTIVITY',
 'PARENTING',
 'WEATHER',
 'VIDEO_PLAYERS',
 'NEWS_AND_MAGAZINES',
 'MAPS_AND_NAVIGATION']

In [10]:
explore_data(apple,0,1,False)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']




In [11]:
explore_data(google,0,1,False)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']




In [12]:
google[10473]

['Life Made WI-Fi Touchscreen Photo Frame',
 '1.9',
 '19',
 '3.0M',
 '1,000+',
 'Free',
 '0',
 'Everyone',
 '',
 'February 11, 2018',
 '1.0.19',
 '4.0 and up']

* DATA CLEARING NOW:

In [13]:
#remove given errors
del google[10473]

In [14]:
#find given duplicates
for app in google:
    name=app[0]
    if name=="Instagram":
        print(app)

['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


In [53]:
unique_values=[]
duplicate_values=[]
for app in google:
    name=app[0]
    if name in unique_values:
        duplicate_values.append(name)
    else:
        unique_values.append(name)

print('Number of duplicates:',len(duplicate_values))
print('\n')
print('Examples of duplicate apps:', duplicate_values[:15])

Number of duplicates: 1181


Examples of duplicate apps: ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack', 'FreshBooks Classic', 'Insightly CRM', 'QuickBooks Accounting: Invoicing & Expenses', 'HipChat - Chat Built for Teams', 'Xero Accounting Software']


* ONLY MOST RECENT DATA OF DUPLICATES WOULD BE LEFT 
(criterion is number of views / review amount)

In [35]:
reviews_max={}
for app in google[1:]:
    name=app[0]
    n_reviews=app[3]
    if name in reviews_max and reviews_max[name]<n_reviews:
        reviews_max[name]=n_reviews
    elif name not in reviews_max:
        reviews_max[name]=n_reviews

len(reviews_max)      #deleting duplicates based on n of reviews, iterates over the rows and picks the one 
                      #with higher amount of views out of existing duplicates

9660

In [44]:
google_clean=[]
already_added=[]
for app in google[1:]:
    name=app[0]
    if name not in already_added:
        google_clean.append(app)
        already_added.append(name)

explore_data(google_clean,0,5)
  

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up']


Number of rows: 9660
Number of columns: 13


* CHECKING THE RESULT (IF THERE IS NO DUPLICATES AND APPLICATION REVIEWS ARE MOST UP TO DATE)

In [54]:
for app in google_clean:
    name=app[0]
    if name=="Instagram":
        print(app)

['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


In [56]:
def english_check(string):
    for i in string:
        if ord(i)>127:
            return False
    return True

* How many rows/apps remained after removing non-english ones:

In [58]:
google_clean_en=[]          #remaining apps that titles are in English
for app in google_clean:
    name=str(app[0])
    if english_check(name):
        google_clean_en.append(app)
        
explore_data(google_clean_en,1,5)

['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up']


['Paper flowers instructions', 'ART_AND_DESIGN', '4.4', '167', '5.6M', '50,000+', 'Free', '0', 'Everyone', 'Art & Design', 'March 26, 2017', '1.0', '2.3 and up']


Number of rows: 9118
Number of columns: 13


* Removing paid apps as the company revenues are based on advertisments therefore views amount are more important than paid apps

In [61]:
google_clean_en_free=[]
for app in google_clean_en:
    price=app[6]
    if price=='Free':
        google_clean_en_free.append(app)

explore_data(google_clean_en_free,0,3)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 8405
Number of columns: 13


* defining function that calculates percentage values of each genre

In [62]:
def frequency(data_set,index):
    frequency_table={}
    total=0
    for app in data_set:
        value=app[index]
        total+=1
        if value in frequency_table:
            frequency_table[value]+=1
        else:
            frequency_table[value]=1

    table_percentages = {}
    for key in frequency_table:
        percentage = (frequency_table[key] / total) * 100
        table_percentages[key] = percentage 
    
    return table_percentages


frequency(google_clean_en_free,1)


{'ART_AND_DESIGN': 0.7019631171921475,
 'AUTO_AND_VEHICLES': 0.9399167162403332,
 'BEAUTY': 0.6305770374776918,
 'BOOKS_AND_REFERENCE': 2.1891731112433073,
 'BUSINESS': 4.711481261154074,
 'COMICS': 0.5710886377156454,
 'COMMUNICATION': 3.224271267102915,
 'DATING': 1.832242712671029,
 'EDUCATION': 1.2730517549077929,
 'ENTERTAINMENT': 1.106484235574063,
 'EVENTS': 0.7138607971445567,
 'FINANCE': 3.7358715050565143,
 'FOOD_AND_DRINK': 1.2016656751933374,
 'HEALTH_AND_FITNESS': 3.1290898274836407,
 'HOUSE_AND_HOME': 0.8209399167162403,
 'LIBRARIES_AND_DEMO': 0.9042236763831053,
 'LIFESTYLE': 3.8905413444378345,
 'GAME': 9.744199881023201,
 'FAMILY': 18.37001784651993,
 'MEDICAL': 3.6287923854848305,
 'SOCIAL': 2.665080309339679,
 'SHOPPING': 2.2486615110053543,
 'PHOTOGRAPHY': 3.010113027959548,
 'SPORTS': 3.2599643069601427,
 'TRAVEL_AND_LOCAL': 2.3081499107674004,
 'TOOLS': 8.554431885782272,
 'PERSONALIZATION': 3.3075550267697795,
 'PRODUCTIVITY': 3.9738251041046992,
 'PARENTING': 0.

* SORTED FREQUENCY TABLE FOR GOOGLE APPS:

In [76]:
def display_table(dataset, index):
    table = frequency(dataset, index)
    table_display = []
    for key in table:
        key_tuple = (table[key], key)
        table_display.append(key_tuple)
        
    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])
    
    
genres_google=display_table(google_clean_en_free,1)


FAMILY : 18.37001784651993
GAME : 9.744199881023201
TOOLS : 8.554431885782272
BUSINESS : 4.711481261154074
PRODUCTIVITY : 3.9738251041046992
LIFESTYLE : 3.8905413444378345
FINANCE : 3.7358715050565143
MEDICAL : 3.6287923854848305
PERSONALIZATION : 3.3075550267697795
SPORTS : 3.2599643069601427
COMMUNICATION : 3.224271267102915
HEALTH_AND_FITNESS : 3.1290898274836407
PHOTOGRAPHY : 3.010113027959548
NEWS_AND_MAGAZINES : 2.795954788816181
SOCIAL : 2.665080309339679
TRAVEL_AND_LOCAL : 2.3081499107674004
SHOPPING : 2.2486615110053543
BOOKS_AND_REFERENCE : 2.1891731112433073
DATING : 1.832242712671029
VIDEO_PLAYERS : 1.7489589530041643
MAPS_AND_NAVIGATION : 1.3563355145746578
EDUCATION : 1.2730517549077929
FOOD_AND_DRINK : 1.2016656751933374
ENTERTAINMENT : 1.106484235574063
AUTO_AND_VEHICLES : 0.9399167162403332
LIBRARIES_AND_DEMO : 0.9042236763831053
HOUSE_AND_HOME : 0.8209399167162403
WEATHER : 0.7971445568114218
EVENTS : 0.7138607971445567
ART_AND_DESIGN : 0.7019631171921475
PARENTING : 

* SORTED FREQUENCY TABLE FOR APPLE APPS:

In [25]:
genres_apple=display_table(apple,11)

Games : 53.65379272020006
Entertainment : 7.4326201722700755
Education : 6.293414837454849
Photo & Video : 4.848569046957488
Utilities : 3.445401500416782
Health & Fitness : 2.5006946373992776
Productivity : 2.4729091414281745
Social Networking : 2.3200889135871074
Lifestyle : 2.0005557099194218
Music : 1.9171992220061127
Shopping : 1.694915254237288
Sports : 1.5837732703528755
Book : 1.5559877743817727
Finance : 1.4448457904973604
Travel : 1.125312586829675
News : 1.0419560989163656
Weather : 1.0002778549597109
Reference : 0.8891358710752986
Food & Drink : 0.875243123089747
Business : 0.7918866351764378
Navigation : 0.6390664073353709
Medical : 0.31953320366768545
Catalogs : 0.13892747985551543
prime_genre : 0.01389274798555154


After generating frequency table for the genre column it is possible to derermine the following:
1. what is the most common genre
2. patterns
3. designation of most used apps

The frequency tables analyzed on the previous screen showed that the App Store is dominated by apps designed for fun, while Google Play shows a more balanced landscape of both practical and fun apps. 

Now i'm going to get an idea about the kind of apps with the most users.

In [26]:
genres_apple=frequency(apple,11)

for genre in genres_apple:
    total=0
    len_genre=1
    
    for app in apple[1:]:
        genre_app=app[11]
        if genre_app==genre:
            n_ratings=float(app[5])
            total+=n_ratings
            len_genre+=1
    avg_n_ratings=total/len_genre
    print(genre, ':',avg_n_ratings)


prime_genre : 0.0
Social Networking : 45228.07142857143
Photo & Video : 14311.274285714286
Games : 13688.452239192338
Music : 28634.525179856115
Reference : 22066.061538461538
Health & Fitness : 9858.403314917126
Weather : 21877.178082191782
Utilities : 6836.25702811245
Travel : 13957.134146341463
Shopping : 18463.983739837397
News : 12843.815789473685
Navigation : 11601.744680851063
Lifestyle : 6119.268965517242
Entertainment : 7519.623134328358
Food & Drink : 13720.828125
Sports : 13904.95652173913
Book : 5080.079646017699
Finance : 10942.438095238096
Education : 2234.2973568281936
Productivity : 8006.346368715083
Business : 4705.5344827586205
Catalogs : 1575.0
Medical : 568.0833333333334


* On average social networking apps have the highest amount of user reviews

Calculating the average number of installs per app genre for the Google Play data set:

In [85]:
categories_google=frequency(google_clean_en_free,1)

for category in categories_google:
    total=0
    len_category=0
    for app in google_clean_en_free:
        category_app=app[1]
        if category_app==category:
            n_installs=app[5]
            n_installs=n_installs.replace('+','')
            n_installs=n_installs.replace(',','')
            total+=float(n_installs)
            len_category += 1
    avg_n_installs=total/len_category
    print(category,':',avg_n_installs)

ART_AND_DESIGN : 1852900.0
AUTO_AND_VEHICLES : 645317.2278481013
BEAUTY : 513151.88679245283
BOOKS_AND_REFERENCE : 8504745.97826087
BUSINESS : 1602958.308080808
COMICS : 880440.625
COMMUNICATION : 36106662.328413285
DATING : 764959.4610389611
EDUCATION : 3021495.3271028036
ENTERTAINMENT : 22563010.752688173
EVENTS : 232885.83333333334
FINANCE : 1348224.9426751593
FOOD_AND_DRINK : 1974937.1386138613
HEALTH_AND_FITNESS : 4263642.1749049425
HOUSE_AND_HOME : 1371193.6376811594
LIBRARIES_AND_DEMO : 674917.2368421053
LIFESTYLE : 1375297.3058103975
GAME : 15723256.825396825
FAMILY : 2582766.391839378
MEDICAL : 119279.81639344263
SOCIAL : 24441088.17857143
SHOPPING : 7307823.2010582015
PHOTOGRAPHY : 18063710.731225297
SPORTS : 3647640.208029197
TRAVEL_AND_LOCAL : 14487541.68041237
TOOLS : 10974693.051460361
PERSONALIZATION : 5027006.791366907
PRODUCTIVITY : 16972497.946107786
PARENTING : 544745.6363636364
WEATHER : 5219216.7164179105
VIDEO_PLAYERS : 25372256.598639455
NEWS_AND_MAGAZINES : 1000

On average, communication apps have the most installs: 36,106,662. This number is heavily skewed up by a few apps that have over one billion installs (WhatsApp, Facebook Messenger, Skype, Google Chrome, Gmail, and Hangouts), and a few others with over 100 and 500 million installs:

In [31]:
for app in google_clean_en_free:
    if app[1] == 'COMMUNICATION' and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'
                                      or app[5] == '100,000,000+'):
        
        print(app[0], ':', app[5])

WhatsApp Messenger : 1,000,000,000+
imo beta free calls and text : 100,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
imo free video calls and chat : 500,000,000+
Skype - free IM & video calls : 1,000,000,000+
Who : 100,000,000+
GO SMS Pro - Messenger, Free Themes, Emoji : 100,000,000+
LINE: Free Calls & Messages : 500,000,000+
Google Chrome: Fast & Secure : 1,000,000,000+
Firefox Browser fast & private : 100,000,000+
UC Browser - Fast Download Private & Secure : 500,000,000+
Gmail : 1,000,000,000+
Hangouts : 1,000,000,000+
Messenger Lite: Free Calls & Messages : 100,000,000+
Kik : 100,000,000+
KakaoTalk: Free Calls & Text : 100,000,000+
Opera Mini - fast web browser : 100,000,000+
Opera Browser: Fast and Secure : 100,000,000+
Telegram : 100,000,000+
Truecaller: Caller ID, SMS spam blocking & Dialer : 100,000,000+
UC Browser Mini -Tiny Fast Private & Secure : 100,000,000+
Viber Messenger : 500,000,000+
WeChat : 100,000,000+
BBM - Free Call

If all the communication apps that have over 100 million installs are removed, the average installs amount would be reduced roughly:

In [32]:
under_100_m=[]
for app in google_clean_en_free:
    n_installs=app[5]
    n_installs=n_installs.replace(',','')
    n_installs=n_installs.replace('+','')
    if app[1]=="COMMUNICATION" and (float(n_installs)< 100000000):
        under_100_m.append(float(n_installs))
sum(under_100_m)/len(under_100_m)

3190672.7276422763


We see the same pattern for the video players category, which is the runner-up with 25,234,606 installs. The market is dominated by apps like Youtube, Google Play Movies & TV, or MX Player. The pattern is repeated for social apps (where we have giants like Facebook, Instagram, Google+, etc.), photography apps (Google Photos and other popular photo editors), or productivity apps (Microsoft Word, Dropbox, Google Calendar, Evernote, etc.).

Again, the main concern is that these app genres might seem more popular than they really are. Moreover, these niches seem to be dominated by a few giants who are hard to compete against.

The game genre seems pretty popular, but previously we found out this part of the market seems a bit saturated, so we'd like to come up with a different app recommendation if possible.

The books and reference genre looks fairly popular as well, with an average number of installs of 8,504,745. It's interesting to explore more, since we found this genre has some potential to work well on the App Store, and our aim is to recommend an app genre that shows potential for being profitable on both the App Store and Google Play.

Let's take a look at some of the apps from this genre and their number of installs:

In [43]:
for app in google_clean_en_free:
    if app[1]=='BOOKS_AND_REFERENCE':
        print(app[0],':',app[5])
    

E-Book Read - Read Book for free : 50,000+
Download free book with green book : 100,000+
Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Free Panda Radio Music : 100,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
English Grammar Complete Handbook : 500,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
Google Play Books : 1,000,000,000+
AlReader -any text book reader : 5,000,000+
Offline English Dictionary : 100,000+
Offline: English to Tagalog Dictionary : 500,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
Recipes of Prophetic Medicine for free : 500,000+
Anonymous caller detection : 10,000+
Ebook Reader : 5,000,000+
Litnet - E-books : 100,000+
Read books online : 5,000,000+
English to Urdu Dictionary : 500,000+
eBoox: book reader fb2 epub zip : 1,000,000+
English Persian Dictionary : 500,000+
Flybook : 500,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
English translation from Bengali : 100,000

The book and reference genre includes a variety of apps: software for processing and reading ebooks, various collections of libraries, dictionaries, tutorials on programming or languages, etc. It seems there's still a small number of extremely popular apps that skew the average:

In [44]:
for app in google_clean_en_free:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000,000+'
                                            or app[5] == '500,000,000+'
                                            or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

Google Play Books : 1,000,000,000+
Bible : 100,000,000+
Amazon Kindle : 100,000,000+
Audiobooks from Audible : 100,000,000+


However, it looks like there are only a few very popular apps, so this market still shows potential. Let's try to get some app ideas based on the kind of apps that are somewhere in the middle in terms of popularity (between 1,000,000 and 100,000,000 downloads):

In [47]:
for app in google_clean_en_free:
    if app[1]=='BOOKS_AND_REFERENCE' and (app[5]=='1000000+' 
                                            or app[5]=='5,000,000+'
                                            or app[5] == '10,000,000+'
                                            or app[5] == '50,000,000+'):
        print(app[0],':',app[5])

Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
FBReader: Favorite Book Reader : 10,000,000+
AlReader -any text book reader : 5,000,000+
Ebook Reader : 5,000,000+
Read books online : 5,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
Moon+ Reader : 10,000,000+
Aldiko Book Reader : 10,000,000+
Dictionary - WordWeb : 5,000,000+
50000 Free eBooks & Free AudioBooks : 5,000,000+
Al-Quran (Free) : 10,000,000+
Al Quran Indonesia : 10,000,000+
Al'Quran Bahasa Indonesia : 10,000,000+
Al Quran : EAlim - Translations & MP3 Offline : 5,000,000+
Quran for Android : 10,000,000+
Dictionary.com: Find Definitions for English Words : 10,000,000+
English Dictionary - Offline : 10,000,000+
Bible KJV : 5,000,000+
NOOK: Read eBooks & Magazines : 10,000,000+
Dictionary : 10,000,000+
Spanish English Translator : 10,000,000+
Dictionary - Merriam-Webster : 10,000,000+
JW Library : 10,000,000+
Oxford Dictionary of English : Free : 10,000,000+
English Hindi Dictionary : 10,000,000+
English to Hindi Diction

This niche seems to be dominated by software for processing and reading ebooks, as well as various collections of libraries and dictionaries, so it's probably not a good idea to build similar apps since there'll be some significant competition.

We also notice there are quite a few apps built around the book Quran, which suggests that building an app around a popular book can be profitable. It seems that taking a popular book (perhaps a more recent book) and turning it into an app could be profitable for both the Google Play and the App Store markets.

However, it looks like the market is already full of libraries, so we need to add some special features besides the raw version of the book. This might include daily quotes from the book, an audio version of the book, quizzes on the book, a forum where people can discuss the book, etc.

# COLCLUSION

In this project, we analyzed data about the App Store and Google Play mobile apps with the goal of recommending an app profile that can be profitable for both markets.

We concluded that taking a popular book (perhaps a more recent book) and turning it into an app could be profitable for both the Google Play and the App Store markets. The markets are already full of libraries, so adding some additional features besides the raw version of the book would be a good idea. This might include daily quotes from the book, an audio version of the book, quizzes on the book, a forum where people can discuss the book, etc.